File: Assets/Scripts/Element.js
#pragma strict
var dis:boolean;
var en:boolean;
var el:boolean;
function Start () {

}

function Update () {
	if(dis)
	{
		renderer.material.color.a -= Time.deltaTime;
		if(renderer.material.color.a < 0.01)
		{
			if(el)
			{
				Destroy(gameObject);
			}
			else{
				ChangeB();
				dis = false;
			}
		}
	}
	if(en)
	{
		renderer.material.color.a += Time.deltaTime;
		if(renderer.material.color.a > 0.99)
		{
			en = false;
		}
	}
}
function Change()
{
	yield WaitForSeconds(1);
	dis = true;
}
function ChangeB()
{
	yield WaitForSeconds(0.5);
	en = true;
	gameObject.Find("Manager").SendMessage("en");
}

File: Assets/Scripts/CSharp/Element.cs
using UnityEngine;
using System.Collections;

public class Element : MonoBehaviour 
{

	public bool dis;
	public bool en;
	public bool el;
	
	
	void Start () 
	{
	
	}
	
	void Update () 
	{
		Color tmpCol;
		
		if(dis)
		{
			tmpCol = renderer.material.color;
			tmpCol.a -= Time.deltaTime;
			renderer.material.color = tmpCol;
			
			if(renderer.material.color.a < 0.01)
			{
				if(el)
				{
					Destroy(gameObject);
				}
				else{
					StartCoroutine(ChangeB());
					dis = false;
				}
			}
		}
		
		if(en)
		{
			tmpCol = renderer.material.color;
			tmpCol.a += Time.deltaTime;
			renderer.material.color = tmpCol;
			
			if(renderer.material.color.a > 0.99)
			{
				en = false;
			}
		}
	}
	
	IEnumerator Change()
	{
		yield return new WaitForSeconds(1);
		dis = true;
	}
	
	IEnumerator ChangeB()
	{
		yield return new WaitForSeconds(0.5f);
		en = true;
		GameObject.Find("Manager").SendMessage("en");
	}
}